File: Playwright1/tests/10a TestAutomationSitePage.js
// import { page } from "@playwright/test";
export class AutomationSitePage{
    
    constructor(page){
        this.page = page;
        this.nameTxtbox = this.page.getByPlaceholder("Enter Name");
        this.emailTxtbox = this.page.locator("#email");
        this.phoneTxtbox = this.page.getByRole("textbox", { name: 'Phone' });
    }

    async launch(){
        await this.page.goto("https://testautomationpractice.blogspot.com/");
    }

    async fillBasicForm(name, email, phone){
        await this.nameTxtbox.clear();
        await this.nameTxtbox.fill(name);
        await this.emailTxtbox.clear();
        await this.emailTxtbox.fill(email);
        await this.phoneTxtbox.clear();
        await this.phoneTxtbox.fill(phone);
        await this.page.waitForTimeout(2000);
    }
}

File: Playwright_CS/PlaywrightTests/10a TestAutomationSitePage.cs
using System.Threading.Tasks;
using Microsoft.Playwright;

public class AutomationSitePage
{
    private readonly IPage _page;
    private readonly ILocator? _nameTxtbox;
    private readonly ILocator? _emailTxtbox;
    private readonly ILocator? _phoneTxtbox;


    public AutomationSitePage(IPage page)
    {
        _page = page;
        _nameTxtbox = page.GetByPlaceholder("Enter Name");
        _emailTxtbox = page.Locator("#email");
        _phoneTxtbox = page.GetByRole(AriaRole.Textbox, new() { Name = "Phone" });
    }

    public async Task Launch()
    {
        await _page.GotoAsync("https://testautomationpractice.blogspot.com/");
    }

    public async Task FillBasicForm(String name, String email, String phone)
    {
        await _nameTxtbox.ClearAsync();
        await _nameTxtbox.FillAsync(name);
        await _emailTxtbox.ClearAsync();
        await _emailTxtbox.FillAsync(email);
        await _phoneTxtbox.ClearAsync();
        await _phoneTxtbox.FillAsync(phone);        
    }
}